### PR TITLE
595 refactor and new key adding flow

### LINKED ISSUE #595: Improve UI for adding new key in editor
There are some confusion and limitations of our current experience.
- Adding key doesn't provide create keys with other formats than jpad.
- Key name is confusing with key path (editable only after saving)
- No way to create a key directly from a specific folder
 
Suggestion (2 steps process):
- First step -> keypath, type (jpad/const/...)
- Second step -> same as edit page

### PR DESCRIPTION
**Pull request by michaelkruglos**
closes #595

### CODE REVIEW

**Comment by michaelkruglos** [comment]
Tests aren't ready yet. Working on it.

**Comment by Yshayy** [comment]
set default format to jpad and default value to string

File: e2e/ui/spec/keys/key-value-type.js [hunk 1]
```diff
@@ -38,7 +33,8 @@ describe('key-value-type', () => {
 
   testCases.forEach(({ valueType, value, expected }) => {
     it(`should convert the type of the jpad to ${valueType}`, () => {
-      Key.setValueType(valueType);
+      Key.add().setKeyFormat('jpad').setValueType(valueType).setName('test').addDetails();
```
Comment (michaelkruglos) [inline e2e/ui/spec/keys/key-value-type.js]: remove this test and the code it tests - it's deprecated by this refactor

File: e2e/ui/utils/Key.js [hunk 2]
```diff
@@ -66,6 +68,16 @@ class Key {
...
     browser.click(dataComp('add-new-key'));
+    browser.waitForVisible(addKeyPage, timeout);
+    return this;
+  }
+
+  clickAddDetails() {
+    browser.click(dataComp('add-key-button'));
+  }
+
+  addDetails() {
```
Comment (michaelkruglos) [inline e2e/ui/utils/Key.js]: change name from addDetails to something more descriptive

File: services/editor/spec/unit/store/ducks/selectedKey-spec.js [hunk 3]
```diff
@@ -350,7 +349,7 @@ describe('selectedKey', async () => {
         expect(pushAction[0]).to.deep.include({ payload: expectedPushPayload });
       });
 
-      it('should not open the new saved key if there was key change while saving', async () => {
+      it.skip('should not open the new saved key if there was key change while saving', async () => {
```
Comment (michaelkruglos) [inline services/editor/spec/unit/store/ducks/selectedKey-spec.js]: remove this test

File: services/editor/src/pages/keys/components/KeyPage/KeyPage.js [hunk 4]
```diff
@@ -26,14 +20,14 @@ const KeyPage = compose(
       return {
         selectedKey: state.selectedKey,
         configKey,
-        isInAddMode: configKey === BLANK_KEY_NAME,
         revision: location.query && location.query.revision,
+        formatSelected: state.selectedKey && state.selectedKey.formatSelected,
```
Comment (michaelkruglos) [inline services/editor/src/pages/keys/components/KeyPage/KeyPage.js]: make it a local state and change its name

File: services/editor/src/store/ducks/ducks-utils/blankKeyDefinition.js [hunk 5]
```diff
@@ -15,16 +15,14 @@ export function createBlankKeyManifest(keyName, implementation = { type: 'file',
   };
 }
 
-export function createBlankJPadKey() {
-  const keyDefSource = {
-    partitions: [],
-    valueType: '',
-    rules: [],
-  };
+export function createSource(valueType = '', rules = [], partitions = []) {
```
Comment (michaelkruglos) [inline services/editor/src/store/ducks/ducks-utils/blankKeyDefinition.js]: createJpadSource is a better name

File: services/editor/src/store/ducks/selectedKey.js [hunk 6]
```diff
@@ -66,20 +72,65 @@ function updateDependentKeys(keyName) {
...
+  if (manifest.implementation.type === 'const') {
     return {
       source: manifest.implementation.value,
-      type: "const",
+      type: 'const',
     };
   }
+}
+
+export function changeKeyFormat(newFormat) {
```
Comment (michaelkruglos) [inline services/editor/src/store/ducks/selectedKey.js]: remove validation in favor or dropdown of predefined vlaues

File: services/editor/src/store/ducks/selectedKey.js [hunk 7]
```diff
@@ -66,20 +72,65 @@ function updateDependentKeys(keyName) {
...
+    dispatch({ type: KEY_FORMAT_CHANGED, payload: newFormat });
+  };
+}
+
+const confirmAddKeyAlert = {
+  title: 'Adding New Key',
+  message: 'Adding new key will discard all your changes.\nDo you want to continue?',
+};
+
+export const addKey = shouldShowConfirmationScreen =>
```
Comment (michaelkruglos) [inline services/editor/src/store/ducks/selectedKey.js]: try to extract the shouldContinue in to a guard

File: services/editor/src/store/ducks/selectedKey.js [hunk 8]
```diff
@@ -92,7 +143,9 @@ export function openKey(key, { revision } = {}) {
     }
 
     if (key === BLANK_KEY_NAME) {
-      dispatch({ type: KEY_OPENED, payload: createBlankJPadKey() });
+      dispatch({ type: KEY_OPENED, payload: createBlankKey() });
```
Comment (michaelkruglos) [inline services/editor/src/store/ducks/selectedKey.js]: change it to create key using parameters and dispatch validation

File: services/editor/src/store/ducks/selectedKey.js [hunk 9]
```diff
@@ -193,14 +246,20 @@ const convertRuleValuesAlert = {
 
 export function updateKeyValueType(keyValueType) {
```
Comment (michaelkruglos) [inline services/editor/src/store/ducks/selectedKey.js]: drop this function, its functionality is deprecated

File: services/editor/src/store/ducks/selectedKey.js [hunk 10]
```diff
@@ -304,16 +368,19 @@ const setValidationHintsVisibility = (validationState, isShown) => {
 
 const handleKeyOpened = (state, { payload: { key, ...keyData } }) => {
   let validation;
+  let formatSelected = false;
```
Comment (michaelkruglos) [inline services/editor/src/store/ducks/selectedKey.js]: this should be inferred from the presence or absence of manifest if possible

File: services/editor/src/utils/keyFormatHelpers.js [hunk 11]
```diff
@@ -0,0 +1,13 @@
+export const validKeyFormats = ['const', 'jpad'];
```
Comment (michaelkruglos) [inline services/editor/src/utils/keyFormatHelpers.js]: belongs in UI

File: services/editor/src/store/ducks/selectedKey.js (hunk 8 above)
Comment (michaelkruglos) [reply services/editor/src/store/ducks/selectedKey.js]: It's going to be too complex

**Pull request merged**